### PR TITLE
ci: add windows to matrix tests

### LINKED ISSUE #140: ci: add windows and macos
after #139 is merged we can add macos as well

see https://github.com/actions/runner-images/issues/2150

need for macos
```
brew install docker colima
```
also see https://github.com/actions/runner-images/issues/6216

implemented #139 and #141

### PR DESCRIPTION
**Pull request by glimchb**
Resolved #140 

Signed-off-by: Boris Glimcher <Boris.Glimcher@emc.com>

### CODE REVIEW

**Comment by glimchb** [comment]
fails on
```
Testing cockroachdb
Unable to find image 'cockroachdb/cockroach:latest' locally
latest: Pulling from cockroachdb/cockroach
docker: no matching manifest for windows/amd64 10.0.20348 in the manifest list entries.
See 'docker run --help'.

Error: exit status 125
Error: Process completed with exit code 1.
```

File: magefiles/test.go [hunk 1]
```diff
@@ -89,6 +90,11 @@ func testImpl(impl string) error {
 		return errors.New("unknown `gokv.Store` implementation")
 	}
 
+	// TODO: until docker images for windows appear, skip those test for windows
```
Comment (philippgille) [inline magefiles/test.go:93]: It's unlikely that there will be Windows specific images, so I was hoping we could either
- Switch the Docker daemon to Linux mode (like [this](https://stackoverflow.com/a/57081502)), but apparently the GitHub Action Windows runners don't support it ([source](https://github.com/orgs/community/discussions/25491))
- Or run Docker in [WSL 2](https://docs.docker.com/desktop/wsl/), but apparently the GitHub Action Windows runners only support WSL 1 ([source](https://github.com/actions/runner-images/issues/1143))

So we're out of luck. Let's see if the situation improves in the future. For now it's already a win to test the other gokv stores that don't rely on services running in Docker :+1:

**Review by philippgille — APPROVED** [review]
Awesome, thank you so much! :+1:

File: magefiles/test.go (hunk 1 above)
Comment (glimchb) [reply magefiles/test.go:93]: I found one for mongo
https://hub.docker.com/_/mongo/tags?page=1&name=windows
but yeah, was exploring both options for above points you made as well

**Pull request merged**